### ISSUE TITLE
CMake still doesn't pass along third-party package information for plugin builds

### ISSUE THREAD

**Issue by J-Gras** [zeek/zeek#3408]
Trying to build https://github.com/zeek/zeek-more-hashes using latest master, it seems I am still running into #3247 on macOS:
```
$ ./configure && make
...
/usr/local/zeek/include/zeek/OpaqueVal.h:11:10: fatal error: 'openssl/md5.h' file not found
#include <openssl/md5.h>
```
As recommended by @awelzel, I am opening this as a new issue.

**Comment by Neverlord** [comment]
I can reproduce it. I'll take a look.

**Comment by awelzel** [comment]
With #3420  this should be fixed for 6.2 and the future, but it's somewhat unreasonable for back-porting to 6.0.

Looking at this again, however, we have the following snippet in Zeek's top-level CMakeLists.txt:

https://github.com/zeek/zeek/blob/6d6a95e761693bad7734b92d395b22dba0e8bacc/CMakeLists.txt#L294-L296

But it's placed before `find_package(OpenSSL)` ever ran.. So wonder if naively moving this down would be an okay 6.0 fix, but immediately remove it with 6.2, too.

```
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -291,9 +291,6 @@ target_include_directories(
     zeek_dynamic_plugin_base
     INTERFACE $<INSTALL_INTERFACE:include> $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
               $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>)
-if (OPENSSL_INCLUDE_DIR)
-    target_include_directories(zeek_dynamic_plugin_base INTERFACE "${OPENSSL_INCLUDE_DIR}")
-endif ()
 target_link_libraries(zeek_dynamic_plugin_base INTERFACE Threads::Threads)
 add_library(Zeek::DynamicPluginBase ALIAS zeek_dynamic_plugin_base)
 set_target_properties(zeek_dynamic_plugin_base PROPERTIES EXPORT_NAME DynamicPluginBase)
@@ -761,6 +758,10 @@ if (NOT BINARY_PACKAGING_MODE)
     set(BUILD_SHARED_LIBS yes)
 endif ()
 
+if (OPENSSL_INCLUDE_DIR)
+    target_include_directories(zeek_dynamic_plugin_base INTERFACE "${OPENSSL_INCLUDE_DIR}")
+endif ()
+

```

**Comment by awelzel** [comment]
I'm closing this. @J-Gras , if you want to re-test. This should be fixed for the upcoming 6.0.3 in the `release/6.0` branch and more properly in `master` going forward.
